gem2rpm/default.spec.erb

```diff
@@ -19,35 +36,28 @@ URL: <%= spec.homepage.gsub('http://', 'https://') %>
 Source0: <%= download_path %>%{gem_name}-%{version}.gem
 
 # start specfile generated dependencies
-Requires: ruby(release)
 <% for req in spec.required_ruby_version -%>
 Requires: ruby<%= " #{req}" unless req.empty? %>
 <% end -%>
-<% for req in spec.required_rubygems_version -%>
-Requires: ruby(rubygems)<%= " #{req}" unless req.empty? %>
+<% for req in spec.required_ruby_version -%>
+BuildRequires: <%= requirement "ruby#{'-devel' unless spec.extensions.empty?}", req %>
 <% end -%>
-<% for d in spec.runtime_dependencies -%>
-<% for req in d.requirement -%>
-Requires: rubygem(<%= d.name %>)<%= " #{req}" unless req.empty? %>
+<% for req in spec.required_rubygems_version -%>
+BuildRequires: <%= requirement 'rubygems-devel', req %>
 <% end -%>
+<% unless spec.extensions.empty? -%>
+<% for dep in runtime_dependencies.virtualize -%>
```

Comment:
This block exists because for gems with extensions we need the runtime dependencies to run the `%check` section.